eng/native/configurecompiler.cmake

```diff
@@ -467,10 +470,6 @@ if (MSVC)
   add_compile_options(/ZH:SHA_256) # use SHA256 for generating hashes of compiler processed source files.
   add_compile_options(/source-charset:utf-8) # Force MSVC to compile source as UTF-8.
 
-  if (CLR_CMAKE_HOST_ARCH_I386)
-    add_compile_options(/Gz)
```

Comment:
Is there a problem with using `/Gz` everywhere?

Reply:
With `/Gz` in shared location, we get the following error in installer x86 build:

```cmd
.\build.cmd -subsetcategory installer -arch x86

...
         "C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x86.Debug\corehost\INSTALL.vcxproj" (rebuild target) (1) ->
         "C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x86.Debug\corehost\ALL_BUILD.vcxproj" (default target) (3:2) ->
         "C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x86.Debug\corehost\cli\comhost\comhost.vcxproj" (default target) (4:2) ->
           C:\Users\adeel\Source\Repos\runtime2\src\installer\corehost\cli\fxr_resolver.h(83,1): error C2664: 'propagate_error_writer_t::propagate_error_writer_t(const propagate_error_writer_t &)': cannot convert argument 1 from 'hostfxr_set_error_writer_fn' to 'propagate_error_writer_t::set_error_writer_fn' (compiling source file C:\Users\adeel\Source\Repos\runtime2\src\installer\corehost\cli\comhost\comhost.cpp) [C:\Users\adeel\Source\Repos\runtime2\artifacts\obj\win-x86.Debug\corehost\cli\comhost\comhost.vcxproj]
```

I think it is because this function requires `__cdecl`?

Reply:
Yes, looks like some places are missing `HOSTPOLICY_CALLTYPE`

Reply:
That also worked, now /Gz is back in shared location: 2264fa0. `__cdecl` is undefined in pal.h, so I just used __cdecl or HOSTPOLICY_CALLTYPE, whichever was available via included headers.